### DISCUSSION TITLE
Torchscript training from Detectron2

### DISCUSSION THREAD

**Discussion by mdhaisne** [facebookresearch/detectron2#3905] [Q&A]
Hello, 
I was wondering why it is not possible today to train a model from detectron2 after an export in scripting / tracing mode ?
In this [page ](https://detectron2.readthedocs.io/en/latest/_modules/detectron2/export/torchscript.html#scripting_with_instances) there is this assert:
```python
assert (
        not model.training
    ), "Currently we only support exporting models in evaluation mode to torchscript"
```
And on this [page](https://detectron2.readthedocs.io/en/latest/_modules/detectron2/export/flatten.html#TracingAdapter) about Tracing Adapter, the forward method is using `with torch.no_grad()`. Maybe it is possible to replace this forward method to enable training ? 

Thanks

**Comment by zhanghang1989** [comment]
torchscript model is used for inference only.
